src/utils/log.rs

```diff
@@ -58,3 +61,56 @@ unsafe extern "C" fn log_callback(importance: wlr_log_importance,
     });
     RUST_LOGGING_FN(importance, message);
 }
+
+pub struct Logger;
+
+impl Logger {
+    pub fn init(level: LevelFilter) {
+        init_logging(
+            match level {
+                LevelFilter::Off => WLR_SILENT,
+                LevelFilter::Warn | LevelFilter::Error => WLR_ERROR,
+                LevelFilter::Info => WLR_INFO,
+                LevelFilter::Debug | LevelFilter::Trace => WLR_DEBUG,
+            },
+            None,
+        );
+
+        let _ = set_boxed_logger(Box::new(Logger)).map(|_| set_max_level(level));
```

Comment:
This will return an error if it is called twice. Can you instead `.expect` that with a detailed error message? I'd rather not just have the first one take but have the second setting be swallowed up as that's confusing.